packages/core/src/messagingProtocols/proteus/ProteusService/PrekeysGenerator/PrekeysGenerator.store.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CoreDatabase} from '../../../../storage/CoreDB';
+
+const STATE_PRIMARY_KEY = 'prekeys_state';
+
+type PrekeysState = {
+  nbPrekeys: number;
+  highestId: number;
+};
+export class PrekeysGeneratorStore {
+  constructor(private readonly db: CoreDatabase) {}
+
+  private async getState(): Promise<PrekeysState> {
+    return (await this.db.get('prekeys', STATE_PRIMARY_KEY)) ?? {nbPrekeys: 0, highestId: 0};
```

Comment:
This would be different. Since the `promise` is aways going to be defined. 
We want to check if the return of the promise is `undefined` or not. Not the promise itself

Reply:
This would be different. Since the `promise` is aways going to be defined. 
We want to check if the return of the promise is `undefined` or not. Not the promise itself